src/libraries/System.Private.CoreLib/src/System/Threading/Thread.OSX.cs

```diff
@@ -0,0 +1,39 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Win32.SafeHandles;
+using System.Diagnostics;
+using System.Runtime;
+using System.Runtime.InteropServices;
+
+namespace System.Threading
+{
+    internal static class ThreadOSX
+    {
+        private static bool CheckEnableAutoreleasePool()
+        {
+#if !CORECLR
+            return false;
+#else
+            bool isEnabled = CLRConfig.GetBoolValue("System.Threading.Thread.EnableAutoreleasePool", out bool isSet);
+            if (!isSet)
+                return false;
+
+            return isEnabled;
+#endif
+        }
+
+        public static bool EnableAutoreleasePool { get; } = CheckEnableAutoreleasePool();
+    }
+
+    public sealed partial class Thread
+    {
+        internal static void AllocateThreadlocalAutoreleasePool()
+        {
+            if (!ThreadOSX.EnableAutoreleasePool)
+                throw new PlatformNotSupportedException();
```

Comment:
Should this be no-op instead of PNSE when the switch is not set? It is the behavior for threadpool.

Reply:
I am hesitant to do this. I think knowing this is enabled as expected is far better than realizing much later that an `NSAutoreleasePool` hasn't been allocated. More than willing to make it a no-op but I'm reluctant to make these sorts of policy decisions no-ops when called.

Reply:
Actually I think I get your suggestion now. I will make this a no-op.